sdk/modelsrepository/azure-iot-modelsrepository/src/main/java/com/azure/iot/modelsrepository/implementation/FileModelFetcher.java

```diff
@@ -0,0 +1,85 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.iot.modelsrepository.implementation;
+
+import com.azure.core.exception.AzureException;
+import com.azure.core.exception.ServiceResponseException;
+import com.azure.core.util.Context;
+import com.azure.core.util.logging.ClientLogger;
+import com.azure.iot.modelsrepository.DtmiConventions;
+import com.azure.iot.modelsrepository.ModelsDependencyResolution;
+import com.azure.iot.modelsrepository.implementation.models.FetchResult;
+import reactor.core.publisher.Mono;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.LinkedList;
+import java.util.Queue;
+
+/**
+ * The {@link FileModelFetcher} is an implementation of {@link ModelFetcher} interface
+ * for supporting local filesystem based model content fetching.
+ */
+class FileModelFetcher implements ModelFetcher {
+
+    private final ClientLogger logger;
+
+    /**
+     * Creates an instance of {@link FileModelFetcher}
+     */
+    public FileModelFetcher(ClientLogger logger) {
```

Comment:
Why is the client logger passed in? A new instance has to be created in this class with this class' name.

Reply:
fixed.